### ISSUE TITLE
Regex match fails where bytestring succeeds

### ISSUE THREAD

**Issue by dopheide-esnet** [zeek/spicy#332]
I've come across an issue where regex is failing w/
1590177125.062602 analyzer error in /Users/dopheide/esnet/spicy/bgp.spicy, line 3: parse error: failed to match regular expression

However, a bytestring of the same data works fine.  I will be attaching the relevant files to reproduce.

OSX 10.15.4
Zeek 3.1.2-dev
spicy spicyc v0.4.0-309 (master 9ad5b97)

**Comment by dopheide-esnet** [comment]
[bgp.tar.gz](https://github.com/zeek/spicy/files/4670598/bgp.tar.gz)

**Comment by bbannier** [comment]
Code in question is
```
module BGP;

const Marker =    /\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF/; # regexp does not match.
# const Marker = b"\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF"; # does match data.

public type Header = unit {
    stuff: Marker;
    length: int16;
    bgp_type: int8;
    data: bytes &eod;
    on %done {
        print self.length, self.bgp_type;
    }
};
```

**Comment by bbannier** [comment]
I filed #348 to check our support for regexp containing byte literals.

**Comment by rsmmr** [comment]
Closed along with #348.
